drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/editor/menu/CellContextMenu.java

```diff
@@ -0,0 +1,144 @@
+
+    private CellContextMenuView view;
+    private Clipboard clipboard;
+    private GuidedDecisionTableView.Presenter activeDecisionTable;
+
+    @Inject
+    public CellContextMenu( final CellContextMenuView view,
+                            final Clipboard clipboard ) {
+        this.view = view;
+        this.clipboard = clipboard;
+    }
+
+    @PostConstruct
+    void setup() {
+        view.init( this );
+    }
+
+    void onDecisionTableSelectedEvent( final @Observes DecisionTableSelectedEvent event ) {
+        final GuidedDecisionTableView.Presenter dtPresenter = event.getPresenter();
+        activeDecisionTable = dtPresenter;
+        initialise();
+    }
+
+    void onDecisionTableSelectionsChangedEvent( final @Observes DecisionTableSelectionsChangedEvent event ) {
+        final GuidedDecisionTableView.Presenter dtPresenter = event.getPresenter();
+        activeDecisionTable = dtPresenter;
+        initialise();
+    }
+
+    @Override
+    public Widget asWidget() {
+        return view.asWidget();
+    }
+
+    @Override
+    public void show( final int mx,
+                      final int my ) {
+        view.show( mx,
+                   my );
+    }
+
+    @Override
+    public void hide() {
+        view.hide();
+    }
+
+    @Override
+    public void initialise() {
+        if ( activeDecisionTable == null || !activeDecisionTable.getAccess().isEditable() ) {
+            disableMenuItems();
+            return;
+        }
+        final List<IGridData.SelectedCell> selections = activeDecisionTable.getView().getModel().getSelectedCells();
+        if ( selections == null || selections.isEmpty() ) {
+            disableMenuItems();
+            return;
+        }
+        enableMenuItems( selections );
+    }
+
+    @Override
+    public void onCut() {
+        if ( activeDecisionTable != null ) {
+            activeDecisionTable.onCut();
+        }
+        hide();
+    }
+
+    @Override
+    public void onCopy() {
+        if ( activeDecisionTable != null ) {
+            activeDecisionTable.onCopy();
+        }
+        hide();
+    }
+
+    @Override
+    public void onPaste() {
+        if ( activeDecisionTable != null ) {
+            activeDecisionTable.onPaste();
+        }
+        hide();
+    }
+
+    @Override
+    public void onDeleteSelectedCells() {
+        if ( activeDecisionTable != null ) {
+            activeDecisionTable.onDeleteSelectedCells();
+        }
+        hide();
+    }
+
+    private void disableMenuItems() {
+        view.enableCutMenuItem( false );
+        view.enableCopyMenuItem( false );
+        view.enablePasteMenuItem( false );
+        view.enableDeleteCellMenuItem( false );
+    }
+
+    private void enableMenuItems( final List<IGridData.SelectedCell> selections ) {
+        final boolean enabled = selections.size() > 0;
+
+        view.enableCutMenuItem( enabled );
+        view.enableCopyMenuItem( enabled );
+        view.enablePasteMenuItem( clipboard.hasData() );
```

Comment:
Nope, "cut", "copy" and "delete" (cells) are based on whether there are any cells selected (i.e. ```selections.size() > 0```); whereas "paste" is based on whether the "clipboard" contains any data. Disabling the menu entirely (e.g. when the table is read-only is handled here: https://github.com/droolsjbpm/drools-wb/pull/185/files#diff-d18e4fc52c91e6b23cf489db4b6ff798R84)